```
22:20:30 hack/agent-cli/agent_cli.go:262:4:warning: ineffective break statement. Did you mean to break out of the outer loop? (SA4011) (megacheck)
22:20:30 syscall.go:38:5:warning: should omit comparison to bool constant, can be simplified to recursive (S1002) (megacheck)
22:20:30 agent.go:1493:17:warning: unnecessary conversion (unconvert)
22:20:30 Makefile:67: recipe for target 'check' failed
22:20:30 make: *** [check] Error 1
22:20:30 Build step 'Execute shell' marked build as failure
```

--- a/hack/agent-cli/README.md
+++ b/hack/agent-cli/README.md
@@ -32,8 +32,10 @@ cc-agent
 
 From a different shell on the host (as root):
 ```
-# ./agent-cli run --ctl=/tmp/hyper.sock --tty=/tmp/tty.sock
+# ./agent-cli run --ctl=/tmp/hyper.sock --tty=/tmp/tty.sock --wait-for-ready
 ```
+
+Do not use the `--wait-for-ready` flag when the `agent-cli` is terminated or disconnected from the `CTL` and `TTY` sockets and the agent is still listening on those sockets. In this case, the agent will not send a `READY` message.
 
 ## Usage
 

--- a/hack/agent-cli/agent_cli.go
+++ b/hack/agent-cli/agent_cli.go
@@ -233,20 +233,21 @@ func monitorStdInLoop(h *hyperstart.Hyperstart, done chan<- bool) error {
 		}
 	}
 
-	done <- true
+	close(done)
+
 	return nil
 }
 
 func monitorTtyOutLoop(h *hyperstart.Hyperstart, done chan<- bool) error {
 	for {
-		msgCh := make(chan *hyperstart.TtyMessage, 1)
-		errorCh := make(chan bool, 1)
+		msgCh := make(chan *hyperstart.TtyMessage)
+		errorCh := make(chan bool)
 
 		go func() {
 			msg, err := h.ReadIoMessage()
 			if err != nil {
 				magicLog("%s\n", err)
-				errorCh <- true
+				close(errorCh)
 				return
 			}
 
@@ -257,7 +258,7 @@ func monitorTtyOutLoop(h *hyperstart.Hyperstart, done chan<- bool) error {
 		case msg := <-msgCh:
 			dumpFrame(*msg)
 		case <-errorCh:
-			done <- true
+			close(done)
 			break
 		}
 	}
@@ -282,7 +283,7 @@ func mainLoop(c *cli.Context) error {
 		return err
 	}
 
-	done := make(chan bool, 1)
+	done := make(chan bool)
 
 	go monitorStdInLoop(h, done)
 	go monitorTtyOutLoop(h, done)

--- a/hack/agent-cli/agent_cli.go
+++ b/hack/agent-cli/agent_cli.go
@@ -267,6 +267,7 @@ func monitorTtyOutLoop(h *hyperstart.Hyperstart, done chan<- bool) error {
 func mainLoop(c *cli.Context) error {
 	ctlSockPath := c.String("ctl")
 	ttySockPath := c.String("tty")
+	waitForReady := c.Bool("wait-for-ready")
 
 	if ctlSockPath == "" || ttySockPath == "" {
 		return fmt.Errorf("Missing socket path: please provide CTL and TTY socket paths")
@@ -279,8 +280,10 @@ func mainLoop(c *cli.Context) error {
 	}
 	defer h.CloseSockets()
 
-	if err := h.WaitForReady(); err != nil {
-		return err
+	if waitForReady {
+		if err := h.WaitForReady(); err != nil {
+			return err
+		}
 	}
 
 	done := make(chan bool)
@@ -313,6 +316,10 @@ func main() {
 					Name:  "tty",
 					Value: "",
 					Usage: "the TTY socket path",
+				},
+				cli.BoolFlag{
+					Name:  "wait-for-ready",
+					Usage: "boolean flag to wait for READY message first",
 				},
 			},
 			Action: func(context *cli.Context) error {

